subgraphs/_reference_/src/sdk/protocols/generic/pool.ts

```diff
@@ -109,8 +116,14 @@ export class Pool {
    * This function will also update the protocol's total value locked based on the change in this pool's.
    */
   refreshTotalValueLocked(): void {
-    const tvl = this.getInputTokenAmountPrice(this.pool.inputTokenBalance);
-    this.setTotalValueLocked(tvl);
+    let totalValueLockedUSD = BIGDECIMAL_ZERO;
+
+    for (let idx = 0; idx < this.pool.inputTokens.length; idx++) {
+      const inputTokenBalanceUSD = this.pool.inputTokenBalancesUSD[idx];
+      totalValueLockedUSD = totalValueLockedUSD.plus(inputTokenBalanceUSD);
+    }
```

Comment:
Or are you supposed to call `setInputTokenBalances()` before you call this? Maybe this function could call that function to reduce the amount of things the dev has to do?

Reply:
I was thinking that we are calling the refreshTotalValueLocked function only when there is an update in the inputTokenBalances, and we are calling setInputTokenBalancesUSD in it. So whenever there is a change in inputTokensSupply we call setInputTokenBalances which sets the inputTokenBalances and then it calls the inputTokenBalancesUSD and then it updates the TVL.
But I understand your point that if a dev calls refreshTotalValueLocked directly then it won't be updated properly.

Reply:
We can call ```setInputTokenBalancesUSD()``` before this call, my only concern is when we call this function from ```setInputTokenBalances()``` then the pricing function will be called twice, and it can reduce the speed of the subgraph. What do you think?

Reply:
I have added ``setInputTokenBalancesUSD()``` before this call. So we don't have stale prices.

Reply:
Maybe it would be best if there is only 1 call that the dev CAN make and it updates the token balance, then fetches prices, and refreshes TVL? That is really the only thing the dev should care about. The other fucntions should rly just be private then.

I think that strikes the balance between us both here

Reply:
Yes, this approach should work. I'll make the changes.